java-backend/src/main/java/org/kframework/backend/java/kil/KItem.java

```diff
@@ -582,13 +587,16 @@ private Term evaluateFunction(KItem kItem, TermContext context) {
                         }
                     }
 
+                    processMatchResult:
                     if (result != null) {
+                        KItemLog.logEvaluated(kItem, result, nestingLevel);
                         kItem.profiler.evalFuncRuleCounter.increment();
                         return result;
                     } else if (owiseResult != null) {
+                        boolean anyRegRuleUnify = false;
                         if (!kItem.isGround()) {
                             if (context.global().stage != Stage.REWRITING) {
-                                return kItem;
+                                break processMatchResult;
```

Comment:
not sure what's going on here. what is the semantics of this break here?

Reply:
It is a refactoring.
Approximate picture:
Before:
```
if (foo) {
  return kItem;
} else {
  stuff;
  return result;
}
return kItem;
```
After:
```
```
ifLabel:
if (foo) {
  break ifLabel;
} else {
  stuff;
  return result;
}
return kItem;
```

I replaced ~2 such return lcoations with breaks. An equivalent return is the next non-logging statement executed after break. The reason is that I can log a message before the return, and not duplicate it in 3 places. (that is, 3 return places before became one now).

Reply:
Java's ability to break to a label on an if is way too much like a goto for my comfort. If you want to make sure something happens prior to return on every code path, this is what a finally block is for. I will look into whether checkstyle has advanced enough to be able to forbid break-to-if (when we first set it up I believe it could not), but for now please replace this with a finally everywhere.

Reply:
It seems there is still no checkstyle check for this. However, it is still something I would like to avoid in the K project as it makes code harder to understand and reason about.